### DISCUSSION TITLE
Use DAO pattern for class naming for DAOs

### DISCUSSION THREAD

**Discussion by iliyan-velichkov** [eclipse/dirigible#3515] [Ideas]
Currently, all DAOs are named *Repository for example Language**Repository**.
I think will be better if they are *DAO like Language**DAO**.
DAO is more widely used. Repository is something spring specific.

Examples:
![image](https://github.com/eclipse/dirigible/assets/5058839/fe663bb2-636e-43d9-8d9c-6c9449864b3c)

**Comment by delchev** [comment]
https://www.baeldung.com/java-dao-vs-repository

**Comment by ThuF** [comment]
I'm converting this issue into discussion, as there are different opinions which term is the more appropriate one.

As @delchev commented:

## Comparing the Two Patterns
- DAO is an abstraction of data persistence. However, a repository is an abstraction of a collection of objects.
- DAO is a lower-level concept, closer to the storage systems. However, Repository is a higher-level concept, closer to the Domain objects.
- DAO works as a data mapping/access layer, hiding ugly queries. However, a repository is a layer between domains and data access layers, hiding the complexity of collating data and preparing a domain object.
- DAO can’t be implemented using a repository. However, a repository can use a DAO for accessing underlying storage.

Also, if we have an anemic domain, the repository will be just a DAO.

Additionally, **the repository pattern encourages a domain-driven design, providing an easy understanding of the data structure for non-technical team members, too.**

**Reply by ThuF** [reply]
So to me the **Repository** terms seems like the more appropriate one in our use cases.
